nova-tt/pause.hpp

```diff
@@ -21,33 +21,44 @@
 #ifndef NOVA_TT_PAUSE_HPP
 #define NOVA_TT_PAUSE_HPP
 
-#ifdef __SSE2__
-#include <emmintrin.h>
+#if defined(__i386__) || defined(_M_IX86) || \
+    defined(__x86_64__) || defined(_M_X64)
+/* Intel */
+#define NOVA_TT_PAUSE
+#include <immintrin.h>
+#elif (defined(__ARM_ARCH_6K__) || \
+       defined(__ARM_ARCH_6Z__) || \
+       defined(__ARM_ARCH_6ZK__) || \
+       defined(__ARM_ARCH_6T2__) || \
+       defined(__ARM_ARCH_7__) || \
+       defined(__ARM_ARCH_7A__) || \
+       defined(__ARM_ARCH_7R__) || \
+       defined(__ARM_ARCH_7M__) || \
+       defined(__ARM_ARCH_7S__) || \
+       defined(__ARM_ARCH_8A__) || \
+       defined(__aarch64__))
+/* ARM */
+/* mnemonic 'yield' is supported from ARMv6k onwards */
+#define NOVA_TT_YIELD
+#else
+/* fallback */
+#warning "architecture does not support yield/pause instruction; falling back to thread yielding."
+#include <thread>
 #endif
 
 namespace nova   {
 namespace detail {
 
-#ifdef __SSE2__
-
 static inline void pause()
 {
+#if defined(NOVA_TT_PAUSE)
```

Comment:
keeping the sse2 check would keep the code correct, though it could reasonable to remove the i386 fallback

Reply:
Note that `__SSE2__` is not defined by all compilers, most notably MSVC.

For x86_64 the check isn't really necessary. For i386 we could check for `__SSE2__` (GNUC) resp. `_M_IX86_FP == 2` (MSVC, see https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170).

Reply:
no strong opinion, though i'd then try to have an explicit check for intel architecture here. (also `NOVA_TT_YIELD` may not be the best term, as it's both a specific arm instruction, but also an os functionality (i.e. `this_thread::yield`). so maybe it's better to check for aarch64 explicitly and then for arm-v7 instead of introducing the helper defines